code/clue.js

```diff
@@ -1,59 +1,326 @@
+  // As well as setting image of the card and flipping the cards, showing description, favroutite weapon
+
+  document.getElementById("killerCard").style.background = mystery.killer.color;
   document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    "killerName"
+  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById("killerImage").classList.toggle("imgflip");
+  document.getElementById("killerImage").src = mystery.killer.image;
+  document.getElementById(
+    "killerDescription"
+  ).innerHTML = `${mystery.killer.description}`;
+  document.getElementById(
+    "killerFavWeapon"
+  ).innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+};
+
+//Function to make sure move knows which card is clicked and start the animation for the appropriate card
+
+const clickCard = (card) => {
+  move(card); 
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//Event handler for clicking card
+document.getElementById('killerCard').addEventListener('click', (e) => {
+  e.preventDefault();
+  clickCard(pickKiller);
+})
+
+// This function will be invoked when you click on the weapon card.
+const pickWeapon = () => {
+  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  mystery.weapon = randomSelector(weapons); 
+  document.getElementById(
+    "weaponName"
+  ).innerHTML = `${mystery.weapon.name}`;
+  document.getElementById("weaponImage").src = mystery.weapon.image;
+  document.getElementById(
+    "weaponWeight"
+  ).innerHTML = `Weight: ${mystery.weapon.weight}`;
+};
+
+//Similar event handler for picking weapon. No image flip occurs on weapon
+document.getElementById('weaponCard').addEventListener('click', (e) => {
+  e.preventDefault();
+  clickCard(pickWeapon);
+})
+
+const pickRoom = () => {
+  // This will randomly select a room from the suspects. And add that to the mystery object.
+  mystery.room = randomSelector(rooms);
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  // document.getElementById("killerCard").style.background = mystery.killer.color;
+  document.getElementById(
+    "roomName"
+  ).innerHTML = `${mystery.room}`;
+  //document.getElementById("killerImage").src = mystery.killer.image;
+};
+
+//Event handler for picking room
+document.getElementById('roomCard').addEventListener('click', (e) => {
+  e.preventDefault();
+  clickCard(pickRoom);
+})
+
+// FUNCTION revealMystery that will be invoked when you click that button. If all cards not clicked will show fading message.
+
+const revealMystery = () => {
+  if ((mystery.killer == null) || (mystery.weapon == null) || (mystery.room == null)){
```

Comment:
You could simplify it like this :
`if (!mystery.killer || !mystery.weapon || !mystery.room) { ... }`
For further information, please reference to https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/

Reply:
Thanks. Changed